### PR TITLE
Even faster approximate initialization

### PR DESCRIPTION
**Pull request by oculus-ruben**
This depends on the other diff, but I tested this with a sample of the SUNCG dataset and it seems to produce okay results.  This just uses `cv2` morphological operations and `numpy` to do some more approximate move map computation instead of trying to do fine-grained collision checking using pure python

### CODE REVIEW

**Comment by ppwwyyxx** [comment]
Looks great! How similar are the results of two different ways of initialization?

**Comment by oculus-ruben** [comment]
Let me try to run some tests and get some numbers in terms of how different the initializations are.  The only thing I did was do some visual inspection of the samples generated and they looked okay (didn't collide with furniture, and were in the correct rooms!)

I'll get back with some sample of numbers here :)  Part of the problem was that it was taking 5-10+ minutes for loading one house, which was a bit weird

**Comment by oculus-ruben** [comment]
I'm still testing this out - I didn't forget about it :) sorry for the delay!

**Comment by oculus-ruben** [comment]
Okay, I saved out the `House.moveMap` that was produced with and without the approximate method here.  It seems pretty close now, but there is one weird thing where it seems like I had to double the radius to actually get a `moveMap` that is similar to the original one.

3837c18e8c16aca3c14059382361791b_approx false
3837c18e8c16aca3c14059382361791b_approx true
383b03e0060134afc9db99160af3b49d_approx false
383b03e0060134afc9db99160af3b49d_approx true
3836ddfaf623875685f900cdd4c6a9f7_approx false
3836ddfaf623875685f900cdd4c6a9f7_approx true

**Comment by oculus-ruben** [comment]
Sorry, I thought the image filenames would be preserved, but that was a bad assumption.  These are the image filenames in the order of what was rendered above (names have the house ID and whether or not the approximate method was used):
'3836ddfaf623875685f900cdd4c6a9f7_approx=False.png'
'3836ddfaf623875685f900cdd4c6a9f7_approx=True.png'
'3837c18e8c16aca3c14059382361791b_approx=False.png'
'3837c18e8c16aca3c14059382361791b_approx=True.png'
'383b03e0060134afc9db99160af3b49d_approx=False.png'
'383b03e0060134afc9db99160af3b49d_approx=True.png'

Note that this was also using a restricted room layout from the `tests/test-samples.py` script (looking @ living rooms)

File: House3D/house.py [hunk 1]
```diff
@@ -581,10 +592,31 @@ def is_door(obj):
...
+                self._updateMovableMap(*roi)
+
+        if approximate:
+            self._adjustApproximateRobotMoveMap()
+
+
+    def _adjustApproximateRobotMoveMap(self):
+        # Here we haven't yet accounted for the robot radius, so do some
+        # approximate accommodation
+        robotGridSize = int(np.rint(self.robotRad * 2 * self.n_row / self.L_det))
```
Comment (oculus-ruben) [inline House3D/house.py]: I ended up changing this line to the following:
```
robotGridSize = int(np.rint(self.robotRad * 4 * self.n_row / self.L_det))
```

**Pull request merged**